runtime/runtime-params-estimator/src/vm_estimator.rs

```diff
@@ -130,6 +135,128 @@ pub fn load_and_compile(
 
 const USING_LIGHTBEAM: bool = cfg!(feature = "lightbeam");
 
+fn measure_contract(
+    vm_kind: VMKind,
+    gas_metric: GasMetric,
+    contract: &ContractCode,
+    cache: Option<&dyn CompiledContractCache>,
+) -> u64 {
+    let start = start_count(gas_metric);
+    let vm_config = VMConfig::default();
+    let result = precompile_contract_vm(vm_kind, &contract, &vm_config, cache);
+    assert!(result.is_ok(), "Compilation failed");
+    let end = end_count(gas_metric, &start);
+    end
+}
+
+/// Returns `(a, b)` - approximation coefficients for formula `a + b * x`
+/// where `x` is is the contract size in bytes. Practically, we compute upper bound
+/// of this approximation, assuming that all contract consists of code only.
+fn precompilation_cost(gas_metric: GasMetric, vm_kind: VMKind) -> (Ratio<i128>, Ratio<i128>) {
+    let workdir = tempfile::Builder::new().prefix("runtime_testbed").tempdir().unwrap();
+    let store = create_store(&get_store_path(workdir.path()));
+    let cache_store = Arc::new(StoreCompiledContractCache { store });
+    let cache: Option<&dyn CompiledContractCache> = Some(cache_store.as_ref());
+
+    let mut xs = vec![];
+    let mut ys = vec![];
+
+    // We use core-contracts, e2f60b5b0930a9df2c413e1460e179c65c8876e3.
+    // File 341191, code 279965, data 56627.
+    let data = include_bytes!("../test-contract/res/lockup_contract.wasm");
+    let contract = ContractCode::new(data.to_vec(), None);
+    xs.push(data.len() as u64);
+    ys.push(measure_contract(vm_kind, gas_metric, &contract, cache));
+
+    // File 257516, code 203545, data 50419.
+    let data = include_bytes!("../test-contract/res/staking_pool.wasm");
+    let contract = ContractCode::new(data.to_vec(), None);
+    xs.push(data.len() as u64);
+    ys.push(measure_contract(vm_kind, gas_metric, &contract, cache));
+
+    // File 135358, code 113152, data 19520.
+    let data = include_bytes!("../test-contract/res/voting_contract.wasm");
+    let contract = ContractCode::new(data.to_vec(), None);
+    xs.push(data.len() as u64);
+    ys.push(measure_contract(vm_kind, gas_metric, &contract, cache));
+
+    // File 124250, code 103473, data 18176.
+    let data = include_bytes!("../test-contract/res/whitelist.wasm");
+    let contract = ContractCode::new(data.to_vec(), None);
+    xs.push(data.len() as u64);
+    ys.push(measure_contract(vm_kind, gas_metric, &contract, cache));
+
```

Comment:
Nit: i'd split out `fn least_squares(xs: &[], ys: &[]) -> (Ratio, Ratio)` to keep pure math independent of wasm stuff. This will also allow us to write a unit-test that, if `least_squeares` is feed with a linear function, it exactly recovers coefficients with zero error. The math looks right, but its fiddly, a test would be a nice assuareance.